src/library/scala/collection/immutable/List.scala

```diff
@@ -132,6 +132,8 @@ sealed abstract class List[+A]
     these
   }
 
+  override def isEmpty: Boolean = false
```

Comment:
It might confuse some people reading the code.. How woulde `this eq Nil` perform?

Reply:
That's probably super quick now that we've got a true static final for:

```
  public static final scala.collection.immutable.Nil$ MODULE$;
    descriptor: Lscala/collection/immutable/Nil$;
    flags: ACC_PUBLIC, ACC_STATIC, ACC_FINAL
```